### PR TITLE
feat: Add private '__embeddedViewMode' API for AppLayout

### PR DESCRIPTION
**Pull request by rubencarvalho**
### Description

This PR adds a private `__embeddedViewMode` API to the `AppLayout` component. This allows the mobile view of the AppLayout to be used in an "headless" mode when embedded in a web view.

Quip tech design proposal: UxKdAETYaEff

### How has this been tested?

Unit test + screenshot test page.

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pages/app-layout/embedded-view-mode.tsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React, { useState } from 'react';
```
Comment (just-boris) [inline pages/app-layout/embedded-view-mode.tsx:3]: How do you use this page? It does not render a route because it does not match `*.page.tsx` pattern

Comment (just-boris) [reply pages/app-layout/embedded-view-mode.tsx:3]: I want to preview this page before approving, but I can't

Comment (rubencarvalho) [reply pages/app-layout/embedded-view-mode.tsx:3]: Oops, error while renaming. Updating.

File: pages/app-layout/embedded-view-mode.page.tsx [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+                  options: [
+                    { label: 'None', value: 0 },
+                    { label: 'Last column', value: 1 },
+                  ],
+                },
+              }}
+            />
+          }
+          header={
+            <Header variant="awsui-h1-sticky" actions={<div style={{ height: '10vh' }} />}>
```
Comment (just-boris) [inline pages/app-layout/embedded-view-mode.page.tsx:120]: Split panel toggle does not work, this header has unexpectedly large height.

Not sure if you did all of it intentionally or not, but this page is confusing

**Review by just-boris — DISMISSED** [review]

File: pages/app-layout/embedded-view-mode.page.tsx (hunk 2 above)
Comment (rubencarvalho) [reply pages/app-layout/embedded-view-mode.page.tsx]: I've simplified the page.

**Review by just-boris — APPROVED** [review]

**Pull request merged**